core/base/document.php

```diff
@@ -1783,6 +1788,59 @@ protected function save_settings( $settings ) {
 	 * @access protected
 	 */
 	protected function print_elements( $elements_data ) {
+		if ( ! Plugin::$instance->experiments->is_feature_active( 'e_element_cache' ) ) {
+			$this->do_print_elements( $elements_data );
+
+			return;
+		}
+
+		$cached_data = $this->get_document_cache();
+
+		if ( false === $cached_data ) {
+			add_filter( 'elementor/element/should_render_shortcode', '__return_true' );
+
+			global $wp_scripts, $wp_styles;
+			$scripts_ignored = $wp_scripts->queue;
+			$styles_ignored = $wp_styles->queue;
+
+			ob_start();
+
+			$this->do_print_elements( $elements_data );
+
+			$scripts_to_queue = array_values( array_diff( $wp_scripts->queue, $scripts_ignored ) );
+			$styles_to_queue = array_values( array_diff( $wp_styles->queue, $styles_ignored ) );
+
+			$cached_data = [
+				'content' => ob_get_clean(),
```

Comment:
Maybe `ob_get_clean()` right after `do_print_elements()`?

Reply:
After reading the code again, I will not change this. The `ob_get_clean()` runs exactly in the building of the array. I don't want to create another variable in the method scope.